test/built-ins/Atomics/wait/was-woken-before-timeout.js

```diff
@@ -0,0 +1,55 @@
+// Copyright (C) 2018 Amal Hussein. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+/*---
+esid: sec-atomics.wait
+description: >
+  Test that Atomics.wait returns the right result when it was awoken before
+  a timeout
+info: |
+  Atomics.wait( typedArray, index, value, timeout )
+
+  2.Let i be ? ValidateAtomicAccess(typedArray, index).
+    ...
+      2.Let accessIndex be ? ToIndex(requestIndex).
+
+      9.If IsSharedArrayBuffer(buffer) is false, throw a TypeError exception.
+        ...
+          3.If bufferData is a Data Block, return false
+
+          If value is undefined, then
+          Let index be 0.
+features: [Atomics, SharedArrayBuffer, TypedArray]
+---*/
+
+var timeout = 20000;
+$262.agent.start(
+    `
+  $262.agent.receiveBroadcast(function (sab) { 
+    var int32Array = new Int32Array(sab);
+    $262.agent.report(Atomics.wait(int32Array, 0, 0, ${timeout}));
+    $262.agent.leaving();
+  })
+  `);
+  
```

Comment:
I don't think `timeout` here shares the same scope -- you will need to declare it inside the agent.